tron/config/schema.py

```diff
@@ -31,6 +33,13 @@ def config_object_factory(name, required=None, optional=None):
     config_class.required_keys = required
     config_class.optional_keys = optional
 
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]):
+        supported_keys = set(required + optional)
+        filtered_data = {k: v for k, v in data.items() if k in supported_keys}
+        return cls(**filtered_data)
```

Comment:
almost forgot: we'll probably wanna figure out a good pattern for figuring out how to adapt data for rollback/rollforward - but i *think* we could probably do something here?

(in slack i thought from_dict was in a class that we subclassed somewhere and would allow us to have per-object from_dict overrides - but i think we can still handle things with this format)

otherwise, maybe we move away from this factory in the future and have a less magic way of making objects :p

Reply:
I think this was the fastest way/ least amount of code changes needed to fix the issue. I agree I think in the future we should def move away from this method